astropy/modeling/functional_models.py

```diff
@@ -572,6 +572,92 @@ def fit_deriv(x, amplitude, x_0, fwhm):
         return [d_amplitude, d_x_0, d_fwhm]
 
 
+class Voigt1D(Fittable1DModel):
+    """
+    One dimensional model for the Voigt profile.
+
+    Parameters
+    ----------
+    pos : float
+        Position of the peak
+    amp_L : float
+        The Lorentzian amplitude
+    fwhm_L : float
+        The Lorentzian full width at half maximum
+    fwhm_G : float
+        The Gaussian full width at half maximum
+
+    See Also
+    --------
+    Gaussian1D, Lorentz1D
+
+    Notes
+    -----
+    Algorithm for the computation taken from
+    McLean, A. B., Mitchell, C. E. J. & Swanston, D. M. Implementation of an
+    efficient analytical approximation to the Voigt function for photoemission
+    lineshape analysis. Journal of Electron Spectroscopy and Related Phenomena
+    69, 125-132 (1994)
+    """
+
+    pos = Parameter()
+    amp_L = Parameter()
+    fwhm_L = Parameter()
+    fwhm_G = Parameter()
```

Comment:
On another note, per #2932 I'm trying to provide "default" values for the parameters in most built-in models.  Many of these defaults are not immediately useful, but some are, and it's convenient for testing.  I think in this case some obvious choices for the first two might be: `x_0 = 0`, `amplitude_L = 1`.

For the other two you could either use just 1, or something like `fwhm_L = 2 / pi`, `fwhm_G = 2*sqrt(2 * ln(2))`  It doesn't much matter.  Whatever looks good :)  It would also be great if, per #3894, you included a sample plot in the docstring.  Something out of your example notebook would be fine.